itest/lnd_sendcoins_utxo_test.go

```diff
@@ -0,0 +1,227 @@
+package itest
+
+import (
+	"testing"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/walletrpc"
+	"github.com/lightningnetwork/lnd/lntest"
+	"github.com/lightningnetwork/lnd/lnwallet"
+	"github.com/stretchr/testify/require"
+)
+
+type sendCoinTestCase struct {
+	// name is the name of the target test case.
+	name string
+
+	// amt specifies the sum to be sent.
+	amt int64
+
+	// shouldFail denotes if we expect the channel opening to fail.
+	shouldFail bool
+
+	// sendAll specifies all funds in the wallet or select utxos should
+	// be sent.
+	sendAll bool
+
+	// selectedCoins are coins selected to be used as inputs for the
+	// transaction.
+	selectedCoins []btcutil.Amount
+
+	// expectedErr is the expected error string returned when sendCoins
+	// fails.
+	expectedErr string
+}
+
+// testSendCoinSelectUtxo tests sendCoins with selected utxos as input.
+func testSendCoinSelectUtxo(ht *lntest.HarnessTest) {
+	alice := ht.NewNode("Alice", nil)
+	defer ht.Shutdown(alice)
+
+	initialCoins := []btcutil.Amount{
+		100_000,
+		50_000,
+		300_000,
+		20_000,
+		1_000,
+	}
+
+	sumCoins := func(coins []btcutil.Amount) int64 {
+		coinSum := int64(0)
+		for _, initialCoin := range coins {
+			coinSum += int64(initialCoin)
+		}
+
+		return coinSum
+	}
+
+	var tcs = []*sendCoinTestCase{
+		{
+			name: "sendCoins with selected utxos",
+			selectedCoins: []btcutil.Amount{
+				50_000,
+				300_000,
+			},
+			amt: 220_000,
+		},
+		{
+			name: "sendCoins with selected utxos, sendAll",
+			selectedCoins: []btcutil.Amount{
+				1_000,
+				300_000,
+			},
+			sendAll: true,
+		},
+		{
+			name:    "sendAll coins in wallet",
+			sendAll: true,
+		},
+		{
+			name: "sendCoins, amount specified, " +
+				"no select outpoints",
+			amt: 200_000,
+		},
```

Comment:
We should add a testcase that checks fractions of selected coins. So from the initial coins, if we select `50_000`, `20_000` and `1_000`, and if we send `70_100`, there should be a change utxo of <900 depending on fees plus the remaining wallet balance.

Reply:
I do not think it is within the scope of the `sendcoins` function to test that. This function delegates that functionality to the `sendOutputs` or the `sendOutputWithInput` all of which calls other functions that call the `NewUnsignedTransaction` function, [which tests for that already](https://github.com/btcsuite/btcwallet/blob/6fe19a472a627b5f3a6ab2eda9ed7d8809d04f2e/wallet/txauthor/author_test.go#L44)

Reply:
I think we can add it though and I have updated the PR in that respect.

Reply:
I would just make the amount 50_400. Should be the same thing?

Reply:
Yes that works.